doc/man7/provider-kdf.pod

```diff
@@ -0,0 +1,325 @@
+     OSSL_FUNC_kdf_newctx(const OSSL_DISPATCH *opf);
+
+B<OSSL_DISPATCH> arrays are indexed by numbers that are provided as
+macros in L<openssl-core_dispatch.h(7)>, as follows:
+
+ OSSL_FUNC_kdf_newctx               OSSL_FUNC_KDF_NEWCTX
+ OSSL_FUNC_kdf_freectx              OSSL_FUNC_KDF_FREECTX
+ OSSL_FUNC_kdf_dupctx               OSSL_FUNC_KDF_DUPCTX
+
+ OSSL_FUNC_kdf_reset                OSSL_FUNC_KDF_RESET
+ OSSL_FUNC_kdf_derive               OSSL_FUNC_KDF_DERIVE
+
+ OSSL_FUNC_kdf_get_params           OSSL_FUNC_KDF_GET_PARAMS
+ OSSL_FUNC_kdf_get_ctx_params       OSSL_FUNC_KDF_GET_CTX_PARAMS
+ OSSL_FUNC_kdf_set_ctx_params       OSSL_FUNC_KDF_SET_CTX_PARAMS
+
+ OSSL_FUNC_kdf_gettable_params      OSSL_FUNC_KDF_GETTABLE_PARAMS
+ OSSL_FUNC_kdf_gettable_ctx_params  OSSL_FUNC_KDF_GETTABLE_CTX_PARAMS
+ OSSL_FUNC_kdf_settable_ctx_params  OSSL_FUNC_KDF_SETTABLE_CTX_PARAMS
+
+A KDF algorithm implementation may not implement all of these functions.
+In order to be a consistent set of functions, at least the following functions
+must be implemented: OSSL_FUNC_kdf_newctx(), OSSL_FUNC_kdf_freectx(),
+OSSL_FUNC_kdf_derive().
+All other functions are optional.
+
+=head2 Context Management Functions
+
+OSSL_FUNC_kdf_newctx() should create and return a pointer to a provider side
+structure for holding context information during a KDF operation.
+A pointer to this context will be passed back in a number of the other KDF
+operation function calls.
+The parameter I<provctx> is the provider context generated during provider
+initialisation (see L<provider(7)>).
+
+OSSL_FUNC_kdf_freectx() is passed a pointer to the provider side KDF context in
+the I<kctx> parameter.
+If it receives NULL as I<kctx> value, it should not do anything other than
+return.
+This function should free any resources associated with that context.
+
+OSSL_FUNC_kdf_dupctx() should duplicate the provider side KDF context in the
+I<kctx> parameter and return the duplicate copy.
+
+=head2 Encryption/Decryption Functions
+
+OSSL_FUNC_kdf_reset() initialises a KDF operation given a provider
+side KDF context in the I<kctx> parameter.
+
+OSSL_FUNC_kdf_derive() performs the KDF operatin.
+The I<kctx> parameter contains a pointer to the provider side context.
+The resulting key should be written to I<key> and the amount of data written
+to I<*keylen>.
+
+=head2 KDF Parameters
+
+See L<OSSL_PARAM(3)> for further details on the parameters structure used by
+these functions.
+
+OSSL_FUNC_kdf_get_params() gets details of parameter values associated with the
+provider algorithm and stores them in I<params>.
+
+OSSL_FUNC_kdf_set_ctx_params() sets KDF parameters associated with the given
+provider side KDF context I<kctx> to I<params>.
+Any parameter settings are additional to any that were previously set.
+
+OSSL_FUNC_kdf_get_ctx_params() gets details of currently set parameter values
+associated with the given provider side KDF context I<kctx> and stores them
+in I<params>.
```

Comment:
This is confusing as the parameters you can get is a completely different set than the ones you can set.

Suggesting:
`OSSL_FUNC_kdf_get_ctx_params() retrieves gettable parameter values
associated with the given provider side KDF context I<kctx> and stores them
in I<params>.`